### PR TITLE
Add initial version of sd-jwt-js

### PR DESCRIPTION
**Pull request by lukasjhan**
## Purpose

Implement sd-jwt standard in typescript: https://www.ietf.org/archive/id/draft-ietf-oauth-selective-disclosure-jwt-06.html

Add tests from sd-jwt-python: https://github.com/openwallet-foundation-labs/sd-jwt-python/tree/main/tests/testcases

**Note** 

- discuss revise README.md

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+```bash
+pnpm run {example_file_name}
+
+#example
+pnpm run complex
+```
+
+### Example lists
+
+- complex : Example of issue, present and verify the complex data.
```
[the remark this answers is not in the document]
Comment (pensivej) [reply examples/README.md:22]: Or did you mean 'comprehensive?'

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+- issue: Example of issue sd-jwt
+- present: Example of present sd-jwt
+- verify: Example of verify sd-jwt
+- validate: Example of validate sd-jwt
+- custom: Example of using custom hasher and salt generator for sd-jwt
```
Comment (pensivej) [inline examples/README.md:27]: I would love to have an example using decoy as well.

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        typ: KB_JWT_TYP,
+        alg,
+      },
+      payload,
+    });
+    await kbJwt.sign(privateKey);
+    return kbJwt;
+  }
+
+  public async issue<Payload extends object>(
```
Comment (pensivej) [inline src/index.ts:62]: Main functions like issue and verify should be separated as individual files to help out navigation

**Review by pensivej — COMMENTED** [review]
Please check my review : )

File: examples/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukasjhan) [reply examples/README.md]: I changed the filename to all.ts, and changed the description

File: src/index.ts (hunk 3 above)
Comment (lukasjhan) [reply src/index.ts:62]: I think it will be needed as there are more files. I will proceed then.

File: examples/README.md (hunk 2 above)
Comment (lukasjhan) [reply examples/README.md]: I added decoy example :)

**Review by zustkeeper — APPROVED** [review]

**Pull request merged**